### PR TITLE
DDF-3641 Adds result count only query support to Intrigue

### PR DESCRIPTION
**Pull request by adimka**
#### What does this PR do?
Enables issuing and display of result count only queries via Intrigue.
#### Who is reviewing?
@rzwiefel @mackncheesiest @Lambeaux 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams/ui
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@andrewkfiedler
@bdeining
#### How should this be tested? (List steps with links to updated documentation)
Ingest data, create a query. Issuing the query normally should return results.
Selecting the result count option from the overflow menu should return the same result count, but not display any results.
#### What are the relevant tickets?
[DDF-3641](https://codice.atlassian.net/browse/DDF-3641)
#### Screenshots (if appropriate)
Regular Query
image

Result Count Button
image

Hit Count Result
image

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.view.js [hunk 1]
```diff
@@ -78,6 +79,9 @@ define([
         handleRun: function(){
             this.model.startSearch();
         },
+        handleHitCount: function(){
+           this.model.getHitCount();
```
Comment (mackncheesiest) [inline catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.view.js]: ✏️  I would prefer not to have a `getHitCount` method and just call `this.model.startSearch({hitCountOnly: true})` here.

`getHitCount` doesn't align with the verbiage used in `startSearch`, so it feels like it's doing an unrelated operation. At first glance sounds like it's just a getter accessing some data that's already been fetched, not performing a new search.

File: catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs [hunk 2]
```diff
@@ -18,6 +18,14 @@
         Run
     </div>
 </div>
+<div class="query-interaction interaction-hit-count" title="Hit Count" data-help="Executes the search returning only the hit count">
+    <div class="interaction-icon">
+        <b>#</b>
+    </div>
+    <div class="interaction-text">
+        Hit Count
```
Comment (mackncheesiest) [inline catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs]: ✏️ All the other options in this menu start with "verb" actions (`Run`, `Stop`, `Search Archived`, `Search Historical`, ...). This should have similar wording.

Also, I personally don't think it should be 2nd option and get in the way of the main three `Run`, `Stop`, `Delete` options.

Comment (adimka) [reply catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs]: I think we can go with `Get Hit Count` no problem. I wanted to avoid being too wordy, but it does stand out like this.

I'm not sure why `Run` is in this menu at all considering we have a run button right next to it.
Same goes for stop since `Run` becomes `Stop` when we run a query.
It might belong first or second if those 2 are gone - I'm curious what @andrewkfiedler has to say about that.

File: catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.view.js (hunk 1 above)
Comment (adimka) [reply catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.view.js]: Yeah that's fine. I did this early on before I knew it was going to be that simple to create a hit count only query.

File: catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs (hunk 2 above)
Comment (rzwiefel) [reply catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs]: Definitely better to have it in both locations. Don't want to force a user to use the buttons especially in case (eventually) someone asks us to allow custom buttons there , better to have all options available in the menu

Comment (bdeining) [reply catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs]: I'm not sure that cramming another button would be worthwhile in this case; the hit count query is a special kind of query much like `Search Historical`.  We don't make buttons for those.

File: catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs [hunk 3]
```diff
@@ -18,6 +18,14 @@
         Run
     </div>
 </div>
+<div class="query-interaction interaction-hit-count" title="Hit Count" data-help="Executes the search returning only the hit count">
+    <div class="interaction-icon">
+        <b>#</b>
```
Comment (lcrosenbu) [inline catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs]: ❗️ Recommend updating this icon to be a "Refresh" icon like https://fontawesome.com/icons/sync-alt?style=solid to be more aligned to the future version of Intrigue (see Teresa for mockups).

File: catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs (hunk 2 above)
Comment (lcrosenbu) [inline catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs]: ❗️ Recommend renaming this menu item to be "Refresh Result Count". The term "hit" is less intuitive to the user.

**Comment by bdeining** [comment]
build now

File: catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs [hunk 4]
```diff
@@ -18,6 +18,13 @@
         Run
     </div>
 </div>
+<div class="query-interaction interaction-refresh-result-count" title="Refresh Result Count" data-help="Executes the search returning only the result count">
```
Comment (brendan-hofmann) [inline catalog/ui/catalog-ui-search/src/main/webapp/component/query-interactions/query-interactions.hbs]: ✏️ punctuation `Executes the search, returning only the result count.`

**Comment by adimka** [comment]
build now

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now
